editor/plugins/tiles/tile_map_layer_editor.cpp

```diff
@@ -3709,32 +3852,77 @@ void TileMapLayerEditor::_advanced_menu_button_id_pressed(int p_id) {
 			}
 		}
 
+		undo_redo->commit_action();
+	} else if (p_id == 1) { // Transform internal TileMap layers into TileMapLayers.
+		ERR_FAIL_COND(edited_layer->get_index_in_tile_map() < 0);
+
+		EditorNode *en = EditorNode::get_singleton();
+		Node *edited_scene_root = en->get_edited_scene();
+		ERR_FAIL_NULL(edited_scene_root);
+
+		// Undo rodo actions.
+		EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+		undo_redo->create_action(TTR("Extract TileMap layers as individual TileMapLayer nodes"));
+
+		TileMap *tile_map = Object::cast_to<TileMap>(edited_layer->get_parent());
+		for (int i = 0; i < tile_map->get_layers_count(); i++) {
+			undo_redo->add_do_method(tile_map, "remove_layer", 0);
+		}
+
+		for (int i = 0; i < tile_map->get_layers_count(); i++) {
+			TileMapLayer *new_layer = tile_map->duplicate_layer_from_internal(i);
+			undo_redo->add_do_method(tile_map, "add_child", new_layer);
+			undo_redo->add_do_method(new_layer, "set_owner", edited_scene_root);
+			undo_redo->add_do_method(new_layer, "set_name", tile_map->get_layer_name(i));
+			undo_redo->add_do_property(new_layer, "tile_set", tile_map->get_tileset());
+			undo_redo->add_undo_method(tile_map, "remove_child", new_layer);
+			undo_redo->add_do_reference(new_layer);
+		}
+
+		List<PropertyInfo> prop_list;
+		tile_map->get_property_list(&prop_list);
+		for (PropertyInfo &prop : prop_list) {
+			undo_redo->add_undo_property(tile_map, prop.name, tile_map->get(prop.name));
+		}
 		undo_redo->commit_action();
 	}
 }
 
 void TileMapLayerEditor::_update_bottom_panel() {
 	const TileMapLayer *edited_layer = _get_edited_layer();
-	if (!edited_layer) {
-		return;
+	Ref<TileSet> tile_set;
+	if (edited_layer) {
+		tile_set = edited_layer->get_tile_set();
 	}
 
-	Ref<TileSet> tile_set = edited_layer->get_effective_tile_set();
+	// Update state labels.
+	no_multi_edit_label->hide();
+	no_layer_label->hide();
+	disabled_or_invisible_layer_label->hide();
+	missing_tileset_label->hide();
+	if (is_multi_node_edit) {
+		no_multi_edit_label->show();
+	} else if (!edited_layer) {
+		no_layer_label->show();
+	} else if (!edited_layer->is_enabled() || !edited_layer->is_visible_in_tree()) {
+		disabled_or_invisible_layer_label->show();
+	} else {
+		missing_tileset_label->set_visible(tile_set.is_null());
+	}
 
-	// Update the visibility of controls.
-	missing_tileset_label->set_visible(tile_set.is_null());
+	// Update tabs visibility.
 	for (TileMapLayerSubEditorPlugin::TabData &tab_data : tabs_data) {
 		tab_data.panel->hide();
 	}
-	if (tile_set.is_valid()) {
+	if (!no_multi_edit_label->is_visible() && !no_layer_label->is_visible() && !disabled_or_invisible_layer_label->is_visible() && !missing_tileset_label->is_visible()) {
 		tabs_data[tabs_bar->get_current_tab()].panel->show();
 	}
```

Comment:
You could merge this and use `set_visible()`.

Reply:
I meant merging it with the loop, but it's fine like that too, I guess.

Reply:
I mean, it was to avoid checking the condition every time. But yeah, I don't think it matters anyway tbh.